### PR TITLE
Add a unit test that checks that the returned FSharpProjectOptions graph doesn't contain duplicates

### PR DESCRIPTION
**Pull request by safesparrow**
...as a proxy for checking that every project is mapped once and cached.

Test creates a "dense" graph:
- 4 layers, 2 projects in each layer
- Projects in each layer reference both projects from the previous layer

It then calls `mapManyOptions` and asserts that the returned graph of `FSharpProjectOptions` contains no duplicates ie. objects are reused.

## Test results
When I removed the caching on the branch, the result was:
```
[23:12:19 ERR] Main tests.When creating FCS options, caches them failed in 00:00:07.4900000.
Mapping should reuse instances of FSharpProjectOptions and only create one per project.
expected: 8
  actual: 52
...
Failed:   1
        Main tests.When creating FCS options, caches them
Errored:  0 <Expecto>
```

When caching is enabled, test passes:
```
Failed:   0
Errored:  0 <Expecto>
```

### CODE REVIEW

File: test/Ionide.ProjInfo.Tests/Tests.fs [hunk 1]
```diff
@@ -824,18 +824,12 @@ let testFCSmapManyProj toolsPath workspaceLoader (workspaceFactory: ToolsPath ->
...
                   | None -> () ]
 
-
         let rec allP2P (po: FSharpProjectOptions) =
             [ for reference in po.ReferencedProjects do
                   let opts = internalGetProjectOptions reference |> Option.get
                   yield reference.OutputFile, opts
                   yield! allP2P opts ]
 
-        let expectP2PKeyIsTargetPath (pos: Map<string, ProjectOptions>) fcsPo =
```
Comment (safesparrow) [inline test/Ionide.ProjInfo.Tests/Tests.fs:834]: Unused
